src/commands/endpoint/select.ts

```diff
@@ -0,0 +1,51 @@
+import { ShellConfig } from "../../lib/config/index";
+import { Args, Command } from "@oclif/core";
+import { searchSelect } from "../../lib/search-select";
+
+export default class DefaultEndpointCommand extends Command {
+  static description = "Set an endpoint as the default one.";
+
+  static examples = [
+    "$ fauna endpoint select",
+    "$ fauna endpoint select endpoint",
+  ];
+
+  static args = {
+    name: Args.string({
+      required: false,
+      description: "New default endpoint",
+    }),
+  };
+
+  static aliases = ["default-endpoint"];
+  static deprecateAliases = true;
+
+  async run() {
+    const config = ShellConfig.read({});
+
+    await this.execute(config);
+  }
+
+  async execute(config: ShellConfig) {
+    const { args } = await this.parse();
+    await this.parse();
+
+    const name =
+      args.name ??
+      (await searchSelect({
+        message: "Select a new default endpoint",
+        choices: Object.keys(config.rootConfig.endpoints).map((endpoint) => ({
```

Comment:
what happens if there are no endpoints here?

Reply:
It won't work. I'll add a check for no endpoints, and error early and say something like "create an endpoin with `fauna cloud-login`"